### DISCUSSION TITLE
如何给method定义统一的返回类型

### DISCUSSION THREAD

**Discussion by songpeng154** [alovajs/alova#311] [Q&A]
### 请详细描述bug

假如后台返回的数据格式统一是这样的
> ```ts
> interface Result<D = any> {
>      code: number
>      msg: string
>      data: D
>}
我如何给method定义一个统一的返回类型

### 复现步骤

- 

### 期望的表现

希望能有一个配置统一的返回类型的地方
>```ts
>alovaInstance.Post<{ list: any[] }>('/test').then(data => {
>    /*
>     * data的类型应该是
>     *  {
>     *      code: number
>     *      msg: string
>     *      data: { list: any[] }
>     *  }
>     * */
>    console.log(data)
>})

### alova 版本

2.18..0

**Comment by JOU-amjs** [comment]
问题请在discussion中开贴。

对于这个问题，推荐在全局响应拦截器中直接返回data数据，然后只要在请求的位置定义data相关的类型即可。
如果因某种原因无法在全局响应拦截器中转换数据，可以定义一个统一的响应类型框架，如下：

```ts
interface Response<R> {
  code: number;
  msg: string;
  data: R;
}

// 使用
alovaInstance.Post<Response<{ list: any[] }>>(...)

```
